### PR TITLE
chore(flags): Add new debugging property `$feature_flag_bootstrapped_response` and `$feature_flag_bootstrapped_payload` to `$feature_flag_called` event

### PR DESCRIPTION
**Pull request by havenbarnes**
## Problem

When SDK users make use of feature flag bootstrapping, the response shown in the $feature_flag_called event can show different values than what would have been fetched remotely. This can lead to confusion when reviewing flag value breakdown results, for example making it seem as if a flag that is turned completely to `true` was evaluated to `false`. These two new event properties are mostly useful for debugging customer issues, but could also be used in product to show users that the response shown came from bootstrapped values, and not /decide

## Changes

Nothing functional, just additional event properties

### Bump level

- [ ] Major
- [x] Minor
- [ ] Patch

### Libraries affected

- [ ] All of them
- [x] posthog-web
- [ ] posthog-node (no server-side bootstrapping)
- [x] posthog-react-native

### Changelog notes

- Add new debugging property `$feature_flag_bootstrapped_response` and `$feature_flag_bootstrapped_payload` to `$feature_flag_called` event

### CODE REVIEW

**Comment by havenbarnes** [comment]
@dmarticus I'm thinking it may be smart to add an additional property here as well, indicated whether flags have been fetched yet or not at the time of the flag check. Something like `$feature_flags_fetched: boolean`. WDYT?

**Comment by dmarticus** [comment]
> @dmarticus I'm thinking it may be smart to add an additional property here as well, indicated whether flags have been fetched yet or not at the time of the flag check. Something like `$feature_flags_fetched: boolean`. Otherwise, we don't have a clear way to know if `$feature_flag_response` matches `$feature_flag_bootstrapped_response` because bootstrapping was done correctly, or if the flags have just been fetched already. WDYT?

I love this idea; seems like it couldn't hurt to add this type of metadata to our flags requests – maybe something like `used_boostrap_value` is more clear than `feature_flags_fetched`, if the goal is to track that bootstrapping was done correctly.  And I think that could be done in a subsequent PR.  But I like this!

**Review by dmarticus — APPROVED** [review]
stamped with a comment!

File: posthog-react-native/CHANGELOG.md
Comment (marandaneto) [inline posthog-react-native/CHANGELOG.md:3]: Should this be a minor since it adds a new feature?

**Comment by marandaneto** [comment]
@havenbarnes would be cool to add a test case for bootstrap flags and payloads, check the file `posthog.featureflags.spec.ts`

File: posthog-react-native/CHANGELOG.md
Comment (havenbarnes) [reply posthog-react-native/CHANGELOG.md:3]: @marandaneto  It's technically just for debugging ability, no plans to use in product right now. You know better here, so let me know what you think!

Comment (dmarticus) [reply posthog-react-native/CHANGELOG.md:3]: eh I think @marandaneto has a good point since it's kinda going to be used in our product (once those events get propagated to the payload, all users will have access to it).  Useful for debugging on our end, sure, but could also be handy for customers I imagine.

Comment (marandaneto) [reply posthog-react-native/CHANGELOG.md:3]: your call here, just food for though, the patch is just for bug fixes usually (semver).

**Comment by havenbarnes** [comment]
Thanks @marandaneto , added some tests. Also added the `$used_bootstrap_value` property discussed above. I'm imitating same behavior as `this.instance.decideEndpointWasHit` from `posthog-js` (https://github.com/PostHog/posthog-js/blob/bootstrapped-flag-evt-property/src/posthog-featureflags.ts#L295). I'll wait for your approval!

File: posthog-react-native/CHANGELOG.md
Comment (havenbarnes) [reply posthog-react-native/CHANGELOG.md]: Makes sense for traditional semver, updated to minor bump!

File: posthog-react-native/CHANGELOG.md
Comment (marandaneto) [inline posthog-react-native/CHANGELOG.md:14]: I believe those changes (the rest of the file) are unintentional since its a nested bullet list, please revert

File: posthog-web/CHANGELOG.md
Comment (marandaneto) [inline posthog-web/CHANGELOG.md:7]: we can also mention `$used_bootstrap_value`

File: posthog-web/CHANGELOG.md
Comment (marandaneto) [inline posthog-web/CHANGELOG.md:3]: this should be a minor as well similarly to RN

File: posthog-core/src/index.ts
Comment (marandaneto) [inline posthog-core/src/index.ts:1268]: I think it's ok having this but you could also just compare the flag (the cached flag - likely not there if decide isn't called, and the value from the bootstrap flags), so you know if it's being used from the bootstrap or not, right?
Since using storage is not super fast, the less we need the better, your call here, but if possible I'd evaluate this at runtime.

File: posthog-react-native/CHANGELOG.md
Comment (havenbarnes) [reply posthog-react-native/CHANGELOG.md]: Weird my VS code was doing this, reverted

File: posthog-core/src/index.ts
Comment (havenbarnes) [reply posthog-core/src/index.ts:1268]: Are you referring to `PostHogPersistedProperty.FeatureFlags` or a platform specific cache?

I was hoping we could avoid this as well, but we store bootstrapped flags in `PostHogPersistedProperty.FeatureFlags` upon initialization, so it would already be there

**Review by marandaneto — APPROVED** [review]

**Comment by havenbarnes** [comment]
> @havenbarnes I think you can merge/release this, that would help debugging #314 o/

Thanks @marandaneto , i was hoping to wait for to get unblocked https://github.com/PostHog/posthog-js/pull/1567 but obviously this will be helpful regardless. If I can't get it unblocked today I'll just merge this

**Pull request merged**
